### PR TITLE
Add Zoom Speed Changer

### LINKED ISSUE #9035: Zoom speed changer
When in the wilderness I hop zoomed out, it takes 4-5 scrolls to zoom in, and mid fight when unfrozen I often have to zoom out which takes 2 scrolls. Increasing my scroll speed in the windows 10 mouse settings has no effect on my scroll speed in runelite/osrs, I also tested programs specifically for changing mouse scroll speed in specific programs but they dont work on runelite/osrs.

**Could a feature be added which allows us to change the scroll speed in runelite?**

### PR DESCRIPTION
**Pull request by rlw0014**
Fixes #9035 , #5904 
Example:
https://youtu.be/KDI0c3bSvkc

### CODE REVIEW

File: runelite-api/src/main/java/net/runelite/api/ScriptID.java [hunk 1]
```diff
@@ -139,6 +139,15 @@
 	 */
 	public static final int DIARY_QUEST_UPDATE_LINECOUNT = 2523;
 
+	/**
+	 * Handles zoom input
+	 * <ul>
+	 * <li> int  zoom value </li>
```
Comment (Nightfirecat) [inline runelite-api/src/main/java/net/runelite/api/ScriptID.java]: Both parameters being labeled as "zoom value" without further information is not helpful.

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ControlFunction.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2018, Jacob M <https://github.com/jacoblairm>
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ControlFunction.java]:
```suggestion
 * Copyright (c) 2019, Jacob M <https://github.com/jacoblairm>
```

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ControlFunction.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+@Getter
+@AllArgsConstructor
+public enum ControlFunction
+{
+	NONE("None"),
+	CONTROL_TO_ZOOM("Hold to zoom"),
+	CONTROL_TO_RESET("Reset zoom");
+
+	private String name;
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ControlFunction.java]:
```suggestion
	private final String name;
```

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomConfig.java [hunk 4]
```diff
@@ -72,14 +72,37 @@ default boolean relaxCameraPitch()
 		return false;
 	}
 
+
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomConfig.java]: Remove this extra newline.
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 5]
```diff
@@ -85,7 +90,7 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 		int[] intStack = client.getIntStack();
 		int intStackSize = client.getIntStackSize();
 
-		if ("scrollWheelZoom".equals(event.getEventName()) && zoomConfig.requireControlDown() && !controlDown)
+		if ("scrollWheelZoom".equals(event.getEventName()) &&  zoomConfig.controlFunction() == ControlFunction.CONTROL_TO_ZOOM && !controlDown)
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: Reorder the args to perform the cheaper check(s) first.

```suggestion
		if (!controlDown && "scrollWheelZoom".equals(event.getEventName()) && zoomConfig.controlFunction() == ControlFunction.CONTROL_TO_ZOOM)
```

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomConfig.java [hunk 6]
```diff
@@ -72,14 +72,37 @@ default boolean relaxCameraPitch()
...
+	@ConfigItem(
+		keyName = "ctrlZoomValue",
+		name = "Reset zoom position",
+		description = "Position of zoom when it is reset",
+		position = 6
+	)
+	default int ctrlZoomValue()
+	{
+		return 600;
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomConfig.java]: I'm not sure this needs to be configurable or not, but it should at least default to the default zoom level which I believe is 512?

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java [hunk 7]
```diff
@@ -104,6 +109,11 @@ public void onScriptCallbackEvent(ScriptCallbackEvent event)
 			return;
 		}
 
+		if ("scrollWheelZoomIncrement".equals(event.getEventName()) && zoomConfig.zoomIncrement() != 25)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]: `25` should probably be a constant somewhere

File: runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java (hunk 5 above)
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/zoom/ZoomPlugin.java]:
```suggestion
		if ("scrollWheelZoom".equals(event.getEventName()) && zoomConfig.controlFunction() == ControlFunction.CONTROL_TO_ZOOM && !controlDown)
```

**Pull request merged**